libraries/AP_BattMonitor/AP_BattMonitor.h

```diff
@@ -44,6 +44,10 @@
 #define AP_BATTMON_FUELLEVEL_ANALOG_ENABLE (BOARD_FLASH_SIZE > 1024)
 #endif
 
+#ifndef AP_BATTMON_SYNTHETIC_CURRENT_ENABLED
+#define AP_BATTMON_SYNTHETIC_CURRENT_ENABLED  !defined(HAL_BUILD_AP_PERIPH)
```

Comment:
Please put this into `chibios_hwdef.py` instead.

```
diff --git a/libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py b/libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py
index 7703d6079a3..906454011f0 100644
--- a/libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py
+++ b/libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py
@@ -2942,6 +2942,11 @@ def add_apperiph_defaults(f):
 #ifndef AP_COMPASS_DIAGONALS_ENABLED
 #define AP_COMPASS_DIAGONALS_ENABLED 0
 #endif
+
+// disable various battery monitor backends:
+#ifndef AP_BATTMON_SYNTHETIC_CURRENT_ENABLED
+#define AP_BATTMON_SYNTHETIC_CURRENT_ENABLED 0
+#endif
 ''')
 
 def add_bootloader_defaults(f):
```

Reply:
done